openlibrary/templates/work_search.html

```diff
@@ -13,29 +13,24 @@
         if k != 'has_fulltext':
             return changequery(page=None,**{k:[i for i in param.get(k, []) if i != v]})
         else:
-            return changequery(page=None,**{k:None})
-
-    def add_track(k):
-        match k:
-            case "has_fulltext":
-                facet = "Ebook"
-            case "author_key":
-                facet = "Author"
-            case "subject_facet":
-                facet = "Subjects"
-            case "person_facet":
-                facet = "People"
-            case "place_facet":
-                facet = "Places"
-            case "time_facet":
-                facet = "Times"
-            case "first_publish_year":
-                facet = "FirstPublished"
-            case "publisher_facet":
-                facet = "Publisher"
-            case "language_facet":
-                facet = "Language"
-        return "SearchFacet|" + facet
+            return changequery(page=None,**{k:None}
+)
+    def add_track(key: str) -> str:
```

Comment:
Sigh it seems like web.py templating also doesn't support types!
```suggestion
    def add_track(key):
```

Reply:
Actually it somehow does!